lib/block_view_test.go

```diff
@@ -14690,3 +15303,1854 @@ func TestUpdateProfileChangeBack(t *testing.T) {
+		require.Contains(err.Error(), RuleErrorNFTBidLessThanMinBidAmountNanos)
+
+		_, _, _, err = _createNFTBid(
+			t, chain, db, params, 10,
+			m1Pub,
+			m1Priv,
+			post1Hash,
+			1,    /*SerialNumber*/
+			1110, /*BidAmountNanos*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorNFTBidLessThanMinBidAmountNanos)
+	}
+
+	// Have m1,m2,m3 make some legitimate bids, including a bid on serial #0.
+	{
+		bidEntries := DBGetNFTBidEntries(db, post1Hash, 1)
+		require.Equal(0, len(bidEntries))
+
+		// m1 --> <post1, #1>
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m1Pub,
+			m1Priv,
+			post1Hash,
+			1,    /*SerialNumber*/
+			1111, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post1Hash, 1)
+		require.Equal(1, len(bidEntries))
+
+		// m1 --> <post1, #0> (This bid can be any amount since it is a blanket bid)
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m1Pub,
+			m1Priv,
+			post1Hash,
+			0,  /*SerialNumber*/
+			10, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post1Hash, 0)
+		require.Equal(1, len(bidEntries))
+
+		// m2: Add a bid from m2 for fun.
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m2Pub,
+			m2Priv,
+			post1Hash,
+			1,    /*SerialNumber*/
+			1112, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post1Hash, 1)
+		require.Equal(2, len(bidEntries))
+
+		// m3: Add a blanket bid from m3 for fun.
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m3Pub,
+			m3Priv,
+			post1Hash,
+			1,    /*SerialNumber*/
+			1113, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post1Hash, 1)
+		require.Equal(3, len(bidEntries))
+	}
+
+	// Accept m3's bid on #1 and m1's blanked bid on #2, weeeee!
+	{
+		// Balance before.
+		m0BalBefore := _getBalance(t, chain, nil, m0Pub)
+		require.Equal(uint64(14958), m0BalBefore)
+
+		// This will accept m3's serial #1 bid.
+		_acceptNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			1,     /*SerialNumber*/
+			m3Pub, /*bidderPkBase58Check*/
+			1113,  /*bidAmountNanos*/
+			"",    /*UnencryptedUnlockableText*/
+		)
+		bidEntries := DBGetNFTBidEntries(db, post1Hash, 1)
+		require.Equal(0, len(bidEntries))
+
+		// This will accept m1's serial #0 bid.
+		_acceptNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			2,     /*SerialNumber*/
+			m1Pub, /*bidderPkBase58Check*/
+			10,    /*bidAmountNanos*/
+			"",    /*UnencryptedUnlockableText*/
+		)
+		bidEntries = DBGetNFTBidEntries(db, post1Hash, 0)
+		require.Equal(0, len(bidEntries))
+
+		// This NFT doesn't have royalties so m0's balance should be directly related to the bids accepted.
+		m0BalAfter := _getBalance(t, chain, nil, m0Pub)
+		require.Equal(m0BalBefore-4+1113+10, m0BalAfter)
+		require.Equal(uint64(16077), m0BalAfter)
+	}
```

Comment:
we need to test minimum bid amount nanos is different for copies and what happens with bids on serial number 0. If a user tries to submit a bid on serial number 0 for less than all NFTs bid amount nanos across all serial numbers, should that bid be rejected?

Reply:
Ah, see my response to your other comment about min bid amount nanos.  I have written the code such that you can accept serial #0 bids regardless of your min bid amount nanos.  I view these bids as "standing offers" that can be accepted at any time by any owner of the specified post hash.

Reply:
See "TestNFTDifferentMinBidAmountSerialNumbers" for a test I added to check that different min bid amounts on different serial numbers of the same NFT behave as expected.